File: vortexdbg-api/src/main/java/com/vortexdbg/PointerNumber.java
package com.vortexdbg;

import com.vortexdbg.pointer.VortexdbgPointer;

public class PointerNumber extends Number {

    private final VortexdbgPointer value;

    public PointerNumber(VortexdbgPointer value) {
        this.value = value;
    }

    @Override
    public int intValue() {
        return this.value == null ? 0 : (int) this.value.toUIntPeer();
    }

    @Override
    public long longValue() {
        return this.value == null ? 0L : this.value.peer;
    }

    @Override
    public float floatValue() {
        throw new AbstractMethodError();
    }

    @Override
    public double doubleValue() {
        throw new AbstractMethodError();
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/PointerNumber.kt
package com.vortexdbg

import com.vortexdbg.pointer.VortexdbgPointer

class PointerNumber(private val value: VortexdbgPointer?) : Number() {

    override fun toInt(): Int {
        return if (this.value == null) 0 else this.value.toUIntPeer().toInt()
    }

    override fun toLong(): Long {
        return if (this.value == null) 0L else this.value.peer
    }

    override fun toFloat(): Float {
        throw AbstractMethodError()
    }

    override fun toDouble(): Double {
        throw AbstractMethodError()
    }

    override fun toByte(): Byte {
        throw AbstractMethodError()
    }

    override fun toShort(): Short {
        throw AbstractMethodError()
    }

    override fun toString(): String {
        return value.toString()
    }
}
